packages/language-server/src/plugins/svelte/features/getSelectionRanges.ts

```diff
@@ -0,0 +1,68 @@
+import { walk } from 'estree-walker';
+import { Position, Range, SelectionRange } from 'vscode-languageserver';
+import { isInTag, positionAt } from '../../../lib/documents';
+import { SvelteDocument } from '../SvelteDocument';
+
+
+type OffsetRange = {
+    start: number;
+    end: number;
+};
+
+export async function getSelectionRange(
+    svelteDoc: SvelteDocument,
+    position: Position,
+) {
+    const { script, style, moduleScript } = svelteDoc;
+    const { ast: { html } } = await svelteDoc.getCompiled();
+    const transpiled = await svelteDoc.getTranspiled();
+    const content = transpiled.getText();
+    const offset = svelteDoc.offsetAt(position);
```

Comment:
The offset here is wrong. The position has to be converted to a generated position first. For that you need to enhance the mappers inside `SvelteDocument` to support `getGeneratedPosition`.